File: java/src/main/java/com/danifoldi/dml/type/DmlCommentableValue.java
package com.danifoldi.dml.type;

public abstract class DmlCommentableValue extends DmlValue {
    private DmlComment comment;

    public void comment(DmlComment comment) {
        this.comment = comment;
    }

    public DmlComment comment() {
        return this.comment;
    }

    public DmlCommentableValue withComment(String comment) {
        this.comment = new DmlComment(comment);
        return this;
    }
}

File: javascript/src/types/DmlCommentableValue.js
class DmlCommentableValue {
  #comment

  comment(comment) {
    if (comment === undefined) {
      return this.#comment
    }
    this.#comment = comment
  }

  withComment(comment) {
    this.#comment = comment
    return this
  }
}

export default DmlCommentableValue
